### DISCUSSION TITLE
js unusedvar

### DISCUSSION THREAD

**Discussion by heart-er** [vuejs/language-tools#744] [General]
In typescript, we can use noUnusedLocals: true to find unused variable, but in JavaScript, we can't find unused variable, so Is there a solution?

**Comment by ghost** [comment]
You can set up ESLint in your project. 
https://github.com/johnsoncodehk/volar/issues/725#issuecomment-976668565 has an example on how such a setup could look like but it uses way more plugins than you need for your case. The example I linked uses:
* **ESLint**
* **Vue plugin for ESLint**
* TypeScript plugin for ESLint
* Stylelint

For your case, only those in bold are necessary

Here are some rules for the linter that will help:
In ESLint (checking for unused vars in `<script>` part of Vue component)
* https://eslint.org/docs/rules/no-unused-vars

In the Vue plugin for ESLint (checking for unused vars in `<template>` part of Vue component:
* https://eslint.vuejs.org/rules/no-unused-vars.html
* https://eslint.vuejs.org/rules/no-unused-components.html

Here's a demo of how the eslint-plugin-vue would behave in a Vue project: https://ota-meshi.github.io/eslint-plugin-vue-demo/

**Comment by johnsoncodehk** [comment]
@heart-er please see @sethidden suggestion, and also checkout https://www.typescriptlang.org/tsconfig#checkJs.
